src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -3177,7 +3183,7 @@ def _do_construct_handler(self, node, parent):
                 # the Loop (but exclude the END DO from this check).
                 names = walk(node.content[:-1], Fortran2003.Name)
                 if construct_name in [name.string for name in names]:
-                    raise NotImplementedError()
+                    raise NotImplementedError("Unsuppored label reference")
```

Comment:
"Unsupported" and perhaps "...reference within DO"?

Reply:
Done